adapters/coap-vertx-base/src/main/java/org/eclipse/hono/adapter/coap/PreSharedKeyCredentialsProvider.java

```diff
@@ -0,0 +1,69 @@
+/**
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 1.0 which is available at
+ * https://www.eclipse.org/legal/epl-v10.html
+ *
+ * SPDX-License-Identifier: EPL-1.0
+ */
+
+package org.eclipse.hono.adapter.coap;
+
+import java.util.Objects;
+
+import org.eclipse.hono.client.HonoClient;
+import org.eclipse.hono.service.auth.device.CredentialsApiAuthProvider;
+import org.eclipse.hono.service.auth.device.DeviceCredentials;
+import org.eclipse.hono.util.CredentialsObject;
+import org.springframework.beans.factory.annotation.Autowired;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+
+/**
+ * Authentication provider supporting pre-shared-key handshake.
+ * 
+ * Extends the {@link CredentialsApiAuthProvider} with a public
+ * {@link #getCredentialsForDevice(PreSharedKeyDeviceIdentity)} to provided the stored secret key for the device. In
+ * difference to the authentication model supported direct by the {@link CredentialsApiAuthProvider}, which compares a
+ * already available key with the stored one, a pre-shared-key handshake requires to access the secret key to generate a
+ * proper finish message.
+ */
+public final class PreSharedKeyCredentialsProvider extends CredentialsApiAuthProvider {
```

Comment:
I am not sure I understand the purpose of this class. I would have simply created an implementation of Scandium's PskStore that uses a CredentialsClient to retrieve the pre-shared key from the Credentials service.